### PR TITLE
created a throttle utility function

### LINKED ISSUE #69: Create throttle Utility Function
### `throttle`

Creates a throttled function that only invokes `func` at most once per `wait` milliseconds.

**File**: `src/utils/function/throttle/throttle.ts`
**Test File**: `src/utils/function/throttle/throttle.test.ts`

**Parameters**:
- `func: Function`: The function to throttle.
- `wait: number`: The number of milliseconds to throttle.

**Returns**:
- `Function`: Returns the new throttled function.

**Example**:
```ts
const throttledFunc = throttle(() => console.log('Hello'), 300);
throttledFunc();
```

### PR DESCRIPTION
**Pull request by NawinKumarSharma**
Closes #69 

## Summary by CodeRabbit

- **New Features**
	- Introduced a `throttle` function to optimize the rate of function execution, including a cancellation method.
	- Added an `isNaN` utility function to accurately check for NaN values.

- **Tests**
	- Implemented a comprehensive suite of unit tests for the `throttle` function, covering various scenarios to ensure its robustness.

These enhancements improve performance and reliability in handling function calls and type checking within the application.

### CODE REVIEW

**Review by bharat-dussa — CHANGES_REQUESTED** [review]

**Comment by bharat-dussa** [comment]
@NawinKumarSharma  Thank you for your contribution. 
We have some urgency with this function, I have updated with covering most of edge cases and along with test cases. 

Please look into the PR to get more Insights!.

Rest Assured!
This PR Still counts as `hacktoberfest`. 

Best of luck for rest of the PR's.


Happy Coding!...
Keep Typing.... 😄

File: src/utils/function/throttle/throttle.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+export function throttle<T extends (...args: unknown[]) => unknown>(
+  func: T,
+  wait: number
+): T & throttleCancel {
+  if (typeof func !== "function") {
+    throw new TypeError("Expected a function as the first argument");
+  }
+
+  if (typeof wait !== "number" || wait < 0) {
+    throw new RangeError("Wait time must be a positive number");
```
[the remark this answers is not in the document]
Comment (bharat-dussa) [reply src/utils/function/throttle/throttle.ts:27]: Why some may pass as 0 right

**Review by bharat-dussa — APPROVED** [review]

**Review by bharat-dussa — APPROVED** [review]
